app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -0,0 +1,404 @@
+        routeLineApi.updateTraveledRouteLine(point).let {
+            routeLineView.renderRouteLineUpdate(
+                binding.mapView.getMapboxMap().getStyle()!!,
+                it
+            )
+        }
+    }
+
+    /**
+     * The SDK triggers [NavigationRouteAlternativesObserver] when available alternatives change.
+     */
+    private val alternativesObserver = object : NavigationRouteAlternativesObserver {
+        override fun onRouteAlternatives(
+            routeProgress: RouteProgress,
+            alternatives: List<NavigationRoute>,
+            routerOrigin: RouterOrigin
+        ) {
+            // Set the suggested alternatives
+            val updatedRoutes = mutableListOf<NavigationRoute>()
+            updatedRoutes.add(routeProgress.navigationRoute) // only primary route should persist
+            updatedRoutes.addAll(alternatives) // all old alternatives should be replaced by the new ones
+            mapboxNavigation.setNavigationRoutes(updatedRoutes)
+        }
+
+        override fun onRouteAlternativesError(error: RouteAlternativesError) {
+            // no impl
+        }
+    }
+
+    /**
+     * Gets notified with progress along the currently active route.
+     */
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeLineApi.updateWithRouteProgress(routeProgress) { result ->
+            binding.mapView.getMapboxMap().getStyle()?.apply {
+                routeLineView.renderRouteLineUpdate(this, result)
+            }
+        }
+    }
+
+    /**
+     * This is one way to keep the route(s) appearing on the map in sync with
+     * MapboxNavigation. When this observer is called the route data is used to draw route(s)
+     * on the map.
+     */
+    private val routesObserver = RoutesObserver { result ->
+        lifecycleScope.launch {
+            routeLineApi.setNavigationRoutes(
+                newRoutes = result.navigationRoutes,
+                alternativeRoutesMetadata = mapboxNavigation.getAlternativeMetadataFor(
+                    result.navigationRoutes
+                )
+            ).apply {
+                routeLineView.renderRouteDrawData(
+                    binding.mapView.getMapboxMap().getStyle()!!,
+                    this
+                )
+            }
+        }
+    }
+
+    /**
+     * Click on any point of the alternative route on the map to make it primary.
+     */
+    private val mapClickListener = OnMapClickListener {
+        lifecycleScope.launch {
+            routeLineApi.findClosestRoute(
+                it,
+                binding.mapView.getMapboxMap(),
+                routeClickPadding
+            ) {
+                val routeFound = it.value?.navigationRoute
+                if (routeFound != null && routeFound != routeLineApi.getPrimaryNavigationRoute()) {
+                    val reOrderedRoutes = routeLineApi.getNavigationRoutes()
+                        .filter { it != routeFound }
```

Comment:
let's replace `it` by some name to increase readability. Examples are for users who use a feature for a first time